Sources/TensorFlow/Optimizers/SGD.swift

```diff
@@ -14,8 +14,73 @@
 
 /// Stochastic gradient descent (SGD) optimizer.
 ///
-/// An optimizer that implements stochastic gradient descent, with support for momentum, learning
-/// rate decay, and Nesterov momentum.
+/// Implements the stochastic gradient descent optimizer algorithm with support 
+/// for momentum, learning rate decay, and Nesterov momentum. Momentum and 
+/// Nesterov momentum (a.k.a. the Nesterov accelerated gradient method) are 
+/// first-order optimization methods that can improve the training speed and the
+/// convergence rate of gradient descent.
+///
+/// Reference: ["A Stochastic Approximation Method"](
+/// https://projecteuclid.org/euclid.aoms/1177729586) (Robbins and Monro, 1951), 
+/// ["On the Stochastic Approximation Method of Robbins and Monro"](
+/// https://projecteuclid.org/euclid.aoms/1177729391) (Wolfowitz, 1952) and
+/// ["Stochastic Estimation of the Maximum of a Regression Function"](
+/// https://projecteuclid.org/euclid.aoms/1177729392) (Kiefer and Wolfowitz, 1952), 
+/// ["Some methods of speeding up the convergence of iteration method"](
+/// https://vsokolov.org/courses/750/2018/files/polyak64.pdf) (Polyak, 1964),
+/// ["A method for unconstrained convex minimization problem with the rate of 
+/// convergence"](http://mpawankumar.info/teaching/cdt-big-data/nesterov83.pdf) 
+/// (Nesterov, 1983).
+/// 
+/// - Parameters:
+///     - learningRate: A float. The learning rate (default value: 0.01).
+///     - momentum: The momentum factor. It accelerates stochastic gradient 
+///     descent in the relevant direction and dampens oscillations (default 
+///     value: 0).
+///     - decay: A float. The learning rate decay (default value: 0).
+///     - nesterov: A boolean. The Nesterov’s accelerated gradient method. 
+///     (default value: true).
+///
+/// ### Examples: ###
+///  
+/// // - Transfer learning from pre-trained VGG19 weights:
+/// 
+/// ````
+/// ...
+/// // Define the base model architecture of VGG19.
```

Comment:
Hi, We really like your improvement of the docs, but we find that the example makes the docs too long. We've been thinking of pushing an update to extract the example as an executable example, but haven't found time to get to it. If you're interested in doing this work, that would be great. Thanks.